node-notes-cli/app.js

```diff
@@ -0,0 +1,57 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+let data = await readFile('data.json', 'utf8');
+data = JSON.parse(data);
+const notesObj = data.notes;
+
+async function rewrite() {
+  await writeFile('data.json', JSON.stringify(data, null, 2));
+}
+
+async function read() {
+  for (const [key, value] of Object.entries(notesObj)) {
+    console.log(`${key}: ${value}`);
+  }
+}
+
+async function add() {
+  const id = data.nextId;
+  data.nextId++;
+  notesObj[id] = process.argv[3];
+  rewrite();
+}
+
+async function poof() {
+  if (notesObj[process.argv[3]]) {
```

Comment:
Here's a case where putting `process.argv[3]` into a local variable would make the code much more readable.